### PR TITLE
you can refill mags with other mags

### PR DESCRIPTION
**Pull request by BobSlim**
## About The Pull Request

Removes handful requirement from mag refilling.
Adds delay if it is not a handful.

## Why It's Good For The Game

While it is "realistic" that you'd have to juggle around ammo and handfuls and mags, it simply isn't conducive to simulate that in the game environment. This makes refilling mags less of a pain.

## Changelog
:cl:
add: You can refill mags with other mags, now.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammunition.dm [hunk 1]
```diff
@@ -98,6 +94,17 @@ They're all essentially identical when it comes to getting the job done.
 		to_chat(user, "The rounds don't match up. Better not mix them up.")
 		return
 
+	if(!source.current_rounds)
+		to_chat(user, "[source] is empty.")
```
Comment (psykzz) [inline code/modules/projectiles/ammunition.dm:98]:
```suggestion
		to_chat(user, "<span class='warning'>\The [source] is empty.</span>")
```

File: code/modules/projectiles/ammunition.dm [hunk 2]
```diff
@@ -98,6 +94,17 @@ They're all essentially identical when it comes to getting the job done.
 		to_chat(user, "The rounds don't match up. Better not mix them up.")
 		return
 
+	if(!source.current_rounds)
+		to_chat(user, "<span class='warning'>\The [source] is empty.</span>")
+		return
+
+	if(!istype(source, /obj/item/ammo_magazine/handful)) //if it isn't a handful, give a delay.
+		to_chat(user, "<span class='notice'>You start refilling [src] with [source].</span>")
+		if(!do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
```
Comment (psykzz) [inline code/modules/projectiles/ammunition.dm:103]:
```suggestion
		if(!do_after(user, 1.5 SECONDS, TRUE, src, BUSY_ICON_GENERIC))
```

File: code/modules/projectiles/ammunition.dm [hunk 3]
```diff
@@ -97,8 +97,8 @@ They're all essentially identical when it comes to getting the job done.
 	if(!source.current_rounds)
 		to_chat(user, "<span class='warning'>\The [source] is empty.</span>")
 		return
-
-	if(!istype(source, /obj/item/ammo_magazine/handful)) //if it isn't a handful, give a delay.
+//using handfuls; and filling internal mags has no delay.
+	if(!istype(source, /obj/item/ammo_magazine/handful) && !istype(src, /obj/item/ammo_magazine/internal) )
```
Comment (BobSlim) [inline code/modules/projectiles/ammunition.dm:101]: is this hacky?

Comment (psykzz) [reply code/modules/projectiles/ammunition.dm:101]: Have you tested this?
istype is a specific check it wont work for child objects.

You could instead try `ispath(type, /some/type/path`

Comment (BobSlim) [reply code/modules/projectiles/ammunition.dm:101]: It seemingly works, haven't heard any complaints, but I'll swap it around anyway

Comment (psykzz) [reply code/modules/projectiles/ammunition.dm:101]: It may be you never do checks against other types and this is good enough.

**Pull request merged**